webdriver/tests/bidi/script/evaluate/evaluate.py

```diff
@@ -0,0 +1,165 @@
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget, ScriptEvaluateResultException
+from ... import recursive_compare, any_string
+
+
+@pytest.mark.asyncio
+async def test_eval(bidi_session, top_context):
+    result = await bidi_session.script.evaluate(
+        expression="1 + 2",
+        target=ContextTarget(top_context["context"]))
+
+    assert result == {
+        "type": "number",
+        "value": 3}
+
+
+@pytest.mark.asyncio
+async def test_params_expression_invalid_script(bidi_session, top_context):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.evaluate(
+            expression='))) !!@@## some invalid JS script (((',
+            target=ContextTarget(top_context["context"]))
+    recursive_compare({
+        'realm': any_string,
+        'exceptionDetails': {
+            'columnNumber': 0,
+            'exception': {
+                'handle': any_string,
+                'type': 'error'},
+            'lineNumber': 0,
+            'stackTrace': {
+                'callFrames': []},
+            'text': any_string}},
+        exception.value.result)
+
+
+@pytest.mark.asyncio
+async def test_exception(bidi_session, top_context):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.evaluate(
+            expression="throw Error('SOME_ERROR_MESSAGE')",
+            target=ContextTarget(top_context["context"]))
+
+    recursive_compare({
+        'realm': any_string,
+        'exceptionDetails': {
+            'columnNumber': 0,
+            'exception': {
+                'handle': any_string,
+                'type': 'error'},
+            'lineNumber': 0,
+            'stackTrace': {
+                'callFrames': [{
+                    'columnNumber': 6,
+                    'functionName': '',
+                    'lineNumber': 0,
+                    'url': ''}]},
+            'text': any_string}},
+        exception.value.result)
+
+
+@pytest.mark.asyncio
+async def test_interact_with_dom(bidi_session, top_context):
+    result = await bidi_session.script.evaluate(
+        expression="'window.location.href: ' + window.location.href",
+        target=ContextTarget(top_context["context"]))
+
+    assert result == {
+        "type": "string",
+        "value": "window.location.href: about:blank"}
+
+
+@pytest.mark.asyncio
+async def test_resolved_promise_with_wait_promise_false(bidi_session,
```

Comment:
nit: Let's rename those three tests from `with_wait_promise` to `with_await_promise` to match the argument's name